### PR TITLE
Enterprise user can remove Connected Apps

### LINKED ISSUE #11944: Enterprise user can remove Connected Apps
**_All Global work to be tracked in Clockify #11242 Discovery Endpoints - including testing, code review etc
Aus devs to track in #11242 Task within Macdoch Regen Discovery_**

## Description

As an Enterprise User on OFN, I can see a new tab in my Enterprise Settings called [Connected Apps](https://github.com/openfoodfoundation/openfoodnetwork/issues/11718). There I can control when my data is shared with other services / platforms. **I can also remove this permission.**

image

Also consider current design feedback:

- [ ] https://github.com/orgs/openfoodfoundation/projects/10?pane=issue&itemId=47791017

### Design files

- [Prototype](https://www.figma.com/proto/zkyllJnljTZGc0LuhPmW9Y/Enterprises?page-id=0%3A1&type=design&node-id=16-2&viewport=167%2C-161%2C0.42&t=DAhVFGv2wsDR3u1j-1&scaling=min-zoom&starting-point-node-id=16%3A2&show-proto-sidebar=1)
- [Asset files](https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery---Back-Office?type=design&node-id=1073%3A2&mode=design&t=W9ltPUsP0VpDID9s-1)

### Icons

- Toggle off: U-F057 cross circle
- Toggle on: U-F058 tick circle

Next step: 

* Add some instance configuration for Instance manager to enable/disable "Connected App"
* managing "Connected App"
  * First step, have some sort of configuration to add/remove "Connected App"
  * Improvement, create a super admin interface to manage "Connected App", to consider what happen when app is deleted/updated :
     * what happen to enterprise data ? 
     * notify enterprise of changes ?

## Acceptance Criteria (e.g.)

Enterprise User

 1. As an Enterprise User, go to Settings, see tab Connected Apps, 
 2. See option for the Discover Regenerative with blurb describing what the user is agreeing to
 3. Hit the button to enable data sharing, see some sort of loading indication  (ie: spinner)
 4. After a while, see a link to MiniExtension
 5. Hit the button to disable data sharing.
 6. All the above is reversed and the MiniExtensions URL stops working after a few seconds.

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

- Closes #11944 

A [previous pull request](https://github.com/openfoodfoundation/openfoodnetwork/pull/11937) enabled users to connect their enterprise data to another app. This pull request enables the user to disable that connection again.

[Screencast from 2023-12-20 15-37-05.webm](https://github.com/openfoodfoundation/openfoodnetwork/assets/3524483/f22fc92a-9f2a-4c93-8a0d-ff30af09169d)

#### What should we test?

- Activate feature `connected_apps` for user, enterprise or instance.
- Visit enterprise settings.
- Select tab Connected apps
- Click on *Allow data sharing*.
- Observe a loading message.
- Observe a notice manage your listing.
- Click on *Stop sharing*.
- Everything is in its original state again.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [x] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/reflexes/admin/connected_app_reflex.rb [hunk 1]
```diff
@@ -3,10 +3,43 @@
...
+        app, current_user.spree_api_key,
+        channel: SessionChannel.for_request(request),
+      )
+      morph :nothing
+    end
+
+    def destroy
+      authorize! :admin, enterprise
+
+      app = enterprise.connected_apps.first
```
Comment (rioug) [inline app/reflexes/admin/connected_app_reflex.rb:23]: Correct me if I am wrong, currently we can only have one connected app, and it's hardcoded right ? maybe we should have a comment here to clarify that.

**Review by rioug — APPROVED** [review]
Looks good :+1:

File: app/reflexes/admin/connected_app_reflex.rb (hunk 1 above)
Comment (mkllnk) [reply app/reflexes/admin/connected_app_reflex.rb:23]: That's right.

**Comment by mkllnk** [comment]
Hm, David didn't review this one and is back on holiday. We want to test this in production soon. So I'll merge this for the next release.

**Pull request merged**